### PR TITLE
feat(whiteboard): Add infinite whiteboard for presentation slides

### LINKED ISSUE #20201: Infinite whiteboard support
Today, our whiteboard feature enables individuals to interact and engage with each other over slides. Depending on the slide's structure, imagery, or colours, it can be challenging to dig deeper into an exploration of the subject. To mitigate this, we recommend incorporating blank slides at key points in the presentation. Often, even a single blank slide leaves teachers and students desiring more space.

To address this issue, I propose extending the whiteboard area beyond a single screen. As seen in the mockups below, the idea is to introduce a single icon to the toolbar that toggle the presentation area from being contained to the presentation and or provide an infinite canvas. 

Screenshot 2024-05-08 at 2 45 25 PM

Screenshot 2024-05-08 at 2 43 27 PM

One point to note for this initial concept: When the teacher activates the multi-user whiteboard, the default behaviour is to automatically follow the teacher.

I'm excited to share this concept, as it has been a highly requested feature to enhance the whiteboard functionality beyond just single presentations. I look forward to hearing perspectives from the community.

### LINKED ISSUE #20565: Undo action is showing shapes drawn by other user
**Describe the bug**
The shapes drawn by a user can be restored by another user using the undo action.

**Video**
possibleBug

### PR DESCRIPTION
**Pull request by KDSBrowne**
### What does this PR do?
This PR introduces the functionality to enable or disable the infinite whiteboard feature in presentations. Similar to the multi-user mode, this feature is applied per slide. The infinite whiteboard allows users to extend the whiteboard area beyond a single screen, providing more space for exploration and interaction.

infinite-canvas

The feature is behind a new setting attribute `allowInfiniteCanvas`, which allows administrators to control its availability.

### Closes Issue(s)
Closes #20565

Partially closes #20201

### CODE REVIEW

**Comment by antobinary** [comment]
Do we currently allow infinite whiteboard in breakouts?

**Comment by KDSBrowne** [comment]
@antobinary, yeah it's available in the breakouts

**Comment by KDSBrowne** [comment]
The select tool auto moving was fixed in the tldraw fork. https://github.com/bigbluebutton/tldraw/pull/9

File: bigbluebutton-html5/private/config/settings.yml [hunk 1]
```diff
@@ -997,6 +997,7 @@ public:
   whiteboard:
     annotationsQueueProcessInterval: 60
     cursorInterval: 150
+    allowInfiniteCanvas: true
```
Comment (antobinary) [inline bigbluebutton-html5/private/config/settings.yml:1000]:
```suggestion
    allowInfiniteCanvas: false
```

Note, we should commit this to `false` for now. Administrators should intentionally enable this, knowing it's experimental, there is no recording, etc.

**Pull request merged**